SimPEG/dask/electromagnetics/frequency_domain/simulation.py

```diff
@@ -48,203 +176,182 @@ def fields(self, m=None, return_Ainv=False):
-            u_src = f[src, self._solutionType]
-
-            if col_chunks is None:
-                col_chunks = int(
-                    np.ceil(
-                        float(self.survey.nD)
-                        / np.ceil(
-                            float(u_src.shape[0])
-                            * self.survey.nD
-                            * 8.0
-                            * 1e-6
-                            / self.max_chunk_size
-                        )
-                    )
-                )
-
-            for rx in src.receiver_list:
-                v = np.eye(rx.nD, dtype=float)
-                n_blocs = np.ceil(u_src.shape[1] * rx.nD / col_chunks)
-
-                for block in np.array_split(v, n_blocs, axis=1):
-                    if block.shape[1] == 0:
-                        continue
-
-                    block_count += block.shape[1] * u_src.shape[1]
-                    blocks_dfduT.append(
-                        array.from_delayed(
-                            dfduT(src, rx, self.mesh, f, block),
-                            dtype=np.float32,
-                            shape=(u_src.shape[0], block.shape[1] * u_src.shape[1]),
-                        )
-                    )
-                    blocks_dfdmT.append(
-                        dfdmT(src, rx, self.mesh, f, block),
-                    )
-                    sources.append(src)
-
-                    if block_count >= (col_chunks):
-                        count = parallel_block_compute(
-                            self,
-                            A_i,
-                            Jmatrix,
-                            freq,
-                            f,
-                            sources,
-                            blocks_dfduT,
-                            blocks_dfdmT,
-                            count,
-                            m_size,
-                            u_src.shape,
-                            self._solutionType,
-                        )
-                        blocks_dfduT = []
-                        blocks_dfdmT = []
-                        sources = []
-                        block_count = 0
-
-        if blocks_dfduT:
-            count = parallel_block_compute(
-                self,
-                A_i,
-                Jmatrix,
-                freq,
-                f,
-                sources,
-                blocks_dfduT,
-                blocks_dfdmT,
-                count,
-                m_size,
-                u_src.shape,
-                self._solutionType,
+    fields_array = delayed(f[:, self._solutionType])
+    fields = delayed(f)
+    survey = delayed(self.survey)
+    mesh = delayed(self.mesh)
+    addresses = []
+    blocks_receiver_derivs = []
+
+    for block in blocks:
+        blocks_receiver_derivs.append(
+            receiver_derivs(
+                survey,
+                mesh,
+                fields,
+                block,
             )
+        )
+
```

Comment:
cleanup?

Reply:
If this is useful for debugging, performance tracking, maybe introduce a debug flag?